### PR TITLE
 Fixes for NXP/iMX6

### PR DESCRIPTION
**Pull request by by-gnome**
- update linux config for 5.9
- fix kodi build
- fix IP firewall

### CODE REVIEW

File: projects/NXP/devices/iMX6/options [hunk 1]
```diff
@@ -17,7 +17,7 @@
     KERNEL_TARGET="zImage"
 
   # kernel serial console
-    EXTRA_CMDLINE="console=ttymxc0,115200 console=tty0"
+    EXTRA_CMDLINE="console=ttymxc0,115200 console=tty0 video=HDMI-A-1:1920x1080@60"
```
Comment (jernejsk) [inline projects/NXP/devices/iMX6/options:20]: why is that needed? imx6 is capable of autodetecting native screen resolution.

File: packages/linux/package.mk [hunk 2]
```diff
@@ -148,11 +150,16 @@ pre_make_target() {
     # copy some extra firmware to linux tree
     mkdir -p $PKG_BUILD/external-firmware
       cp -a $(get_build_dir kernel-firmware)/.copied-firmware/{amdgpu,amd-ucode,i915,radeon,e100,rtl_nic} $PKG_BUILD/external-firmware
+      cp -a $(get_build_dir intel-ucode)/intel-ucode $PKG_BUILD/external-firmware
```
Comment (jernejsk) [inline packages/linux/package.mk:153]: This change is unrelated to imx6.

Comment (jernejsk) [reply packages/linux/package.mk:153]: ah, sorry, I see that you only moved it from below. Still, why don't you move this dependency to project options?

**Comment by lrusak** [comment]
You shouldn't need to build the firmware into the kernel. I've never needed to do that for i.MX6 before. What is the reason for doing so?

File: packages/linux/package.mk (hunk 2 above)
Comment (by-gnome) [reply packages/linux/package.mk:153]: SDMA firmware should be built-in, I did it the same way as it is done for x86_64.

Comment (lrusak) [reply packages/linux/package.mk:153]: should be or needs to be?

**Comment by by-gnome** [comment]
@lrusak i.MX SDMA support is configured as built-in and if we don't make 
SDMA firmware built-in we will see in the log:
```
imx-sdma 20ec000.sdma: Direct firmware load for imx/sdma/sdma-imx6q.bin failed with error -2
imx-sdma 20ec000.sdma: external firmware not found, using ROM firmware
```
yes, firmware won't load, but SDMA will work. I'm sorry but I don't like this.
Well if I configure i.MX SDMA support as a module at least one or both audio 
devices (imx6-wandboard-sgtl5000, imx-spdif) in my Wandboard won't work.

File: packages/linux/package.mk (hunk 2 above)
Comment (by-gnome) [reply packages/linux/package.mk:153]: I explained below.

File: packages/linux/package.mk [hunk 3]
```diff
@@ -148,11 +150,16 @@ pre_make_target() {
...
+      cp -a $(get_build_dir kernel-firmware)/imx $PKG_BUILD/external-firmware
+  fi
 
-    cp -a $(get_build_dir intel-ucode)/intel-ucode $PKG_BUILD/external-firmware
-
+  if [ -d $PKG_BUILD/external-firmware/ ]; then
     FW_LIST="$(find $PKG_BUILD/external-firmware \( -type f -o -type l \) \( -iname '*.bin' -o -iname '*.fw' -o -path '*/intel-ucode/*' \) | sed 's|.*external-firmware/||' | sort | xargs)"
-    sed -i "s|CONFIG_EXTRA_FIRMWARE=.*|CONFIG_EXTRA_FIRMWARE=\"${FW_LIST}\"|" $PKG_BUILD/.config
+    sed -i -e "s|^CONFIG_EXTRA_FIRMWARE=.*$|CONFIG_EXTRA_FIRMWARE=\"${FW_LIST}\"|" $PKG_BUILD/.config
+    sed -i -e "/CONFIG_EXTRA_FIRMWARE_DIR/d" -e "/CONFIG_EXTRA_FIRMWARE=.../a CONFIG_EXTRA_FIRMWARE_DIR=\"external-firmware\"" $PKG_BUILD/.config
```
Comment (lrusak) [inline packages/linux/package.mk:162]: why are you changing this? This should be set in the kernel config

File: packages/linux/package.mk [hunk 4]
```diff
@@ -148,11 +150,16 @@ pre_make_target() {
     # copy some extra firmware to linux tree
     mkdir -p $PKG_BUILD/external-firmware
       cp -a $(get_build_dir kernel-firmware)/.copied-firmware/{amdgpu,amd-ucode,i915,radeon,e100,rtl_nic} $PKG_BUILD/external-firmware
+      cp -a $(get_build_dir intel-ucode)/intel-ucode $PKG_BUILD/external-firmware
+  elif [ "$TARGET_ARCH" = "arm" -a "$DEVICE" = "iMX6" ]; then
+    mkdir -p $PKG_BUILD/external-firmware
+      cp -a $(get_build_dir kernel-firmware)/imx $PKG_BUILD/external-firmware
```
Comment (lrusak) [inline packages/linux/package.mk:156]: why not use the sdma firmware from the firmware-imx package? you might as well build in all the firmware then.

File: packages/linux-firmware/firmware-imx/package.mk [hunk 5]
```diff
@@ -22,7 +22,7 @@ makeinstall_target() {
   mkdir -p $INSTALL/$(get_full_firmware_dir)/imx/sdma
     cp -P firmware/sdma/sdma-imx6q.bin $INSTALL/$(get_full_firmware_dir)/imx/sdma
 
-  mkdir -p $INSTALL/$(get_full_firmware_dir)/vpu
```
Comment (lrusak) [inline packages/linux-firmware/firmware-imx/package.mk:25]: you don't really need to change this, the kernel looks for the vpu dir.

https://github.com/torvalds/linux/blob/f01c30de86f1047e9bae1b1b1417b0ce8dcd15b1/drivers/media/platform/coda/coda-common.c#L3057-L3062

Comment (by-gnome) [reply packages/linux-firmware/firmware-imx/package.mk:25]: Yes you are right, I just wanted to get rid of:
```
coda 2040000.vpu: Direct firmware load for vpu_fw_imx6q.bin failed with error -2
coda 2040000.vpu: Using fallback firmware vpu/vpu_fw_imx6q.bin
```

File: packages/linux/package.mk (hunk 4 above)
Comment (by-gnome) [reply packages/linux/package.mk:156]: Yes, maybe firmware-imx will do better here. I'll think about it.

File: packages/linux-firmware/firmware-imx/package.mk (hunk 5 above)
Comment (lrusak) [reply packages/linux-firmware/firmware-imx/package.mk:25]: ok, I can live with that change.

File: projects/NXP/devices/iMX6/options (hunk 1 above)
Comment (lrusak) [reply projects/NXP/devices/iMX6/options:20]: Yes please remove this change. You can add this to your own device if you need this.

File: packages/linux/package.mk (hunk 3 above)
Comment (by-gnome) [reply packages/linux/package.mk:162]: I think that CONFIG_EXTRA_FIRMWARE_DIR should be set exact where CONFIG_EXTRA_FIRMWARE is set.
If we want (this can be in theory) to change the folder name `external-firmware`, we will only need to edit one file.

File: projects/NXP/devices/iMX6/linux/linux.arm.conf [hunk 6]
```diff
@@ -1,17 +1,16 @@
 #
```
Comment (chewitt) [inline projects/NXP/devices/iMX6/linux/linux.arm.conf:1]: ^ chmod -x on the file please

File: projects/NXP/devices/iMX6/options (hunk 1 above)
Comment (by-gnome) [reply projects/NXP/devices/iMX6/options:20]: I get sometimes a resolution of 1024x768, sometimes 1920x1080.
Sometimes I even get 1024x768 on a TV with 1920x1080 in kodi.

IMG_20201114_023138_sz

But perhaps this is due to the problems of reading EDID in my Wandboard.
I should probably remove this and investigate the issue better.

Comment (jernejsk) [reply projects/NXP/devices/iMX6/options:20]: I experienced similar issue on Allwinner images. Workaround is to disable display driver in U-Boot, at least in my case. You can modify U-Boot config via bootloader/config file.

File: projects/NXP/devices/iMX6/linux/linux.arm.conf (hunk 6 above)
Comment (by-gnome) [reply projects/NXP/devices/iMX6/linux/linux.arm.conf:1]: Done.

**Comment by by-gnome** [comment]
I removed commits related to firmware (sdma, vpu, brcm), these changes can be made with another PR. Thanks for your time.

**Review by lrusak — APPROVED** [review]
Thanks for the effort.

We will need to replace the plane usage patch though. I can do that later.

**Review by chewitt — APPROVED** [review]
Having never owned or used iMX6 hardware I don't really have an opinion .. but LGTM :)

**Review by jernejsk — APPROVED** [review]
